base/complex.jl

```diff
@@ -167,6 +167,9 @@ function write(s::IO, z::Complex)
     write(s,real(z),imag(z))
 end
 
+## byte order swaps: real and imaginary part are swapped individually
+bswap(z::Complex) = Complex(bswap(real(z)), bswap(imag(z)))
```

Comment:
`bswap` is defined for hardware floating-point values as well.  I think it is appropriate to define this method for all `Complex{T}`, and let it fail with a `bswap` methoderror if the latter is not defined for `T`.